cluster.go

```diff
@@ -1114,12 +1133,8 @@ func (c *ClusterClient) _processPipeline(ctx context.Context, cmds []Cmder) erro
 				if err == nil {
 					return
 				}
-				if c.opt.ReadOnly && isConnectionError(err) {
-					// TODO maybe not the best place for this? effective tho
-					internal.Logger.Printf(c.Context(), "mark connection as failing %s %s attempt %d", node.String(), err.Error(), attempt)
+				if c.opt.OnErrFunc != nil && !c.opt.OnErrFunc(node, err) {
```

Comment:
Is there no default?

Reply:
yes there is no default in upstream.. changes are only in forked branch

Reply:
this should probably only be on on readonly

Reply:
Honestly this is the least good code. with passing in the node as a stringer. It would be better if i could pass the node itself and you could call mark as failing but it would mean exposing objects more, so not sure which would be more palatable... thoughts?